File: estruturas/1023.py


entradas = [
    '3',
    '3 22',
    '2 11',
    '3 39',
    '5',
    '1 25',
    '2 20',
    '3 31',
    '2 40',
    '6 70',
    '2',
    '1 1',
    '3 2',
    '0'
]

pos = -1
def input():
    global pos
    pos += 1
    return entradas[pos]

num_city = 0
n = int(input())
while n != 0:
    if n == 0:
        break
    
    num_city += 1
    if num_city != 1:
        print()
    total_residents = 0
    total_consumption = 0
    list_consumption = {}
    for i in range(n):
        residents, consumption = map(int, input().split(' '))

        avg_consumption = consumption // residents
        if (avg_consumption in list_consumption.keys()):
            list_consumption[avg_consumption] += residents
        else:
            list_consumption[avg_consumption] = residents

        total_residents += residents
        total_consumption += consumption

    list_sorted = sorted(list_consumption)
    result = ""
    for x in list_sorted:
        result += str(list_consumption[x]) + "-" + str(x) + " "

    print("Cidade# "+str(num_city)+":")
    print(result[:-1])
    print("Consumo medio: %0.2f m3." % (int((total_consumption / total_residents) * 100)/100.0))
    
    n = int(input())

File: estruturas/1023.cpp
#include <iostream>
 
using namespace std;
 
int main() {
    int n, num_city = 0;

    cin >> n;

    while (n != 0) {
        if(num_city != 0) {
            cout << endl;
        }
        int properties[n][2], total_res = 0, total_con = 0, count_array = 0;
        double avg_con;

        cout << "Cidade# " << ++num_city << ":" << endl;

        for(int i = 0; i < n; i++) {
            int x, y;

            cin >> x >> y;

            total_res += x;
            total_con += y;
            int avg_con_res = y/x;
            bool has_inset = false;
            for(int j = 0; j < count_array; j++) {
                if(properties[j][0] == avg_con_res) {
                    properties[j][1] += x;
                    has_inset = true;
                    break;
                } else if(properties[j][0] > avg_con_res) {
                    for(int k = count_array; k > j; k--) {
                        properties[k][0] = properties[k-1][0];
                        properties[k][1] = properties[k-1][1];
                    }
                    properties[j][0] = avg_con_res;
                    properties[j][1] = x;
                    has_inset = true;
                    count_array++;
                    break;
                }
            }
            if(!has_inset) {
                properties[count_array][0] = avg_con_res;
                properties[count_array][1] = x;
                count_array++;
            }
        }
        for(int i = 0; i < count_array; i++) {
            if(((i+1) % count_array) == 0) {
                cout << properties[i][1] << "-" << properties[i][0] << endl;
            } else {
                cout << properties[i][1] << "-" << properties[i][0] << " ";
            }
        }
        avg_con = total_con/float(total_res);
        printf("Consumo medio: %.2lf m3.\n", (int)(avg_con*100)/100.0);
        
        cin >> n;
    }
    
 
    return 0;
}